core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -526,6 +537,19 @@ private[spark] class ExecutorAllocationManager(
       allocationManager.synchronized {
         stageIdToNumTasks(stageId) = numTasks
         allocationManager.onSchedulerBacklogged()
+
+        var numTasksPending = 0
```

Comment:
What situation does that occur in?  When a stage has some tasks with locality preferences and some tasks without?